src/imageLoader/wadors/getPixelData.js

```diff
@@ -29,9 +30,26 @@ function uint8ArrayToString(data, offset, length) {
   return str;
 }
 
-function getPixelData(uri, imageId, mediaType = 'application/octet-stream') {
+function getPixelData(uri, imageId, mediaTypes) {
+  const supportedMediaTypes = {
+    '1.2.840.10008.1.2.5': ['image/x-dicom-rle'],
+    '1.2.840.10008.1.2.4.50': ['image/jpeg'],
+    '1.2.840.10008.1.2.4.51': ['image/jpeg'],
+    '1.2.840.10008.1.2.4.57': ['image/jpeg'],
+    '1.2.840.10008.1.2.4.70': ['image/jpeg', 'image/jll'],
+    '1.2.840.10008.1.2.4.80': ['image/x-jls', 'image/jls'],
+    '1.2.840.10008.1.2.4.81': ['image/x-jls', 'image/jls'],
+    '1.2.840.10008.1.2.4.90': ['image/jp2'],
+    '1.2.840.10008.1.2.4.91': ['image/jp2'],
+    '1.2.840.10008.1.2.4.92': ['image/jpx'],
+    '1.2.840.10008.1.2.4.93': ['image/jpx'],
+  };
```

Comment:
Since this function is running on every request and a study can have +1000 request, I'm over picky here. 

1. you are passing around `mediaTypes` which as I understand is all the media types imported from `mediaTypesUtils/mediaTypes.js` (a constant). Do you really need to do that? can't you just import it inside the `buildMultipart...`? 

2. Also the `supportedMediaTypes` is getting re-created on each request, and that also can get imported. 

Having said that these two are both constants, why are we `buildingMultipart*` again and again? Are you using `getPixelData` in your App directly and you are passing the `mediaTypes` there? I'm afraid this would add a lot of overhead in any app that was working previously

One final question, you say the order of entries in `mediaTypes` are important based on priority, is there a known priority for different types? what criteria did you use to sort it

Reply:
> Since this function is running on every request and a study can have +1000 request, I'm over picky here.
> 
> 1. you are passing around `mediaTypes` which as I understand is all the media types imported from `mediaTypesUtils/mediaTypes.js` (a constant). Do you really need to do that? can't you just import it inside the `buildMultipart...`?
> 2. Also the `supportedMediaTypes` is getting re-created on each request, and that also can get imported.
> 
> Having said that these two are both constants, why are we `buildingMultipart*` again and again? Are you using `getPixelData` in your App directly and you are passing the `mediaTypes` there? I'm afraid this would add a lot of overhead in any app that was working previously
> 

you are right, rebuilding the string everytime makes sense only if the mediaTypes were provided externally, which is not the case here. 

> One final question, you say the order of entries in `mediaTypes` are important based on priority, is there a known priority for different types? what criteria did you use to sort it

the only criterion is that the media type

```
const octetStreamMediaType = 'application/octet-stream';
const octetStreamTransferSyntaxUID = '*';
```

has to be the last (because if it is first, for example the server could decode a jpeg and provide the decoded jpeg). All the others are indepedent and unique (servers will never try to convert a jpeg to a jp2, and so on, since they are very expensive operations).

I also double checked/tested performances and I didn't measured perfomance drawbacks to have one particular mediatype before the others.